File: structural/facad-example.py
class DatabaseConnection:
    def __init__(self):
        pass

    @staticmethod
    def run_query(query):
        return f'QUERY PROCESSED => {query}'


class UserInterface:

    def __init__(self):
        pass

    @staticmethod
    def get_input_data():
        return 'INPUT DATA'

    @staticmethod
    def show_result(result):
        print(result)


class Business:

    def __init__(self):
        pass

    @staticmethod
    def do_a_process():
        data = UserInterface.get_input_data()
        result = DatabaseConnection.run_query(data)
        UserInterface.show_result(result)


if __name__ == "__main__":
    Business.do_a_process()

File: structural/facad-example.js
class DatabaseConnection {
    constructor() {

    }

    static runQuery(query) {
        return 'RUNNING THIS QUERY => ' + query;
    }
}

// ----------------------------------------------------------------------
class UserInterface {
    constructor() {

    }

    static getInputData() {
        return 'INPUT DATA';
    }

    static show_result(result) {
        console.log(result);
    }
}

// ----------------------------------------------------------------------
class Business{
    constructor() {
    }

    static doProcess(){
        let data = UserInterface.getInputData();
        let result = DatabaseConnection.runQuery(data);
        UserInterface.show_result(result);
    }
}

// ----------------------------------------------------------------------
Business.doProcess()
